Drivers: hwinfo: add reset cause shell

alexanderwachter opened the pull request:
This PR adds a shell command to obtain, clear and show available reset causes

alexanderwachter committed:
drivers: hwinfo: shell: Add reset cause shell command

This commit adds shell commands for the reset cause API
- show: show the persistent reset cause
- clear: clear the persistent reset cause
- supported: list all supported reset causes

Signed-off-by: Alexander Wachter <alexander.wachter@leica-geosystems.com>

pdgendt commented on drivers/hwinfo/hwinfo_shell.c:
```diff
@@ -38,8 +38,149 @@ static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_reset_cause,
+	SHELL_CMD_ARG(show, NULL, "Show the persistent reset cause",
+		      cmd_show_reset_cause, 1, 0),
+	SHELL_CMD_ARG(clear, NULL, "Clear all persisten reset causes",
```
```suggestion
	SHELL_CMD_ARG(clear, NULL, "Clear all persistent reset causes",
```

pdgendt commented on drivers/hwinfo/hwinfo_shell.c:
```diff
@@ -38,8 +38,149 @@ static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv
+	}
+}
+
+static void print_all_reset_causes(const struct shell *shell, uint32_t cause)
+{
+	for (uint32_t cause_mask = 1; cause_mask; cause_mask <<= 1) {
+		if ((cause & cause_mask) == 0) {
+			continue;
+		}
+
+		shell_print(shell, "- %s", cause_to_string(cause & cause_mask));
+	}
+}
```
Why not just:
```suggestion
static void print_all_reset_causes(const struct shell *shell, uint32_t cause)
{
	for (uint32_t cause_mask = 1; cause_mask; cause_mask <<= 1) {
		if (cause & cause_mask) {
			shell_print(shell, "- %s", cause_to_string(cause & cause_mask));
		}
	}
}
```

alexanderwachter replied:
Whoops 😅
Much easier, yes!

pdgendt reviewed (approved):
Small nits, otherwise LGTM

alexanderwachter committed:
drivers: hwinfo: shell: rename args to satisfy rule 5.7

This commit changes the shell argument name from shell to sh
to satisfy the warning
"Violation to rule 5.7 (Tag name should be unique) tag: shell"

Signed-off-by: Alexander Wachter <alexander@wachter.cloud>

alexanderwachter commented on drivers/hwinfo/hwinfo_shell.c:
```diff
@@ -19,27 +19,165 @@ static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv
+}
+
+static void print_all_reset_causes(const struct shell *sh, uint32_t cause)
+{
+	for (uint32_t cause_mask = 1; cause_mask; cause_mask <<= 1) {
```
A minimal scope is always preferable.

alexanderwachter commented:
@cfriedt ready to merge :-)

cfriedt commented:
Cc @arvinf
